genesyscloud/architect_emergencygroup/genesyscloud_architect_emergencygroup_proxy.go

```diff
@@ -0,0 +1,124 @@
+		getArchitectEmergencyGroupIdByNameAttr: getArchitectEmergencyGroupIdByNameFn,
+		deleteArchitectEmergencyGroupAttr:      deleteArchitectEmergencyGroupFn,
+	}
+}
+
+func getArchitectEmergencyGroupProxy(clientConfig *platformclientv2.Configuration) *architectEmergencyGroupProxy {
+	if internalProxy == nil {
+		internalProxy = newArchitectEmergencyGroupProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+func (p *architectEmergencyGroupProxy) getAllArchitectEmergencyGroups(ctx context.Context) (*[]platformclientv2.Emergencygroup, *platformclientv2.APIResponse, error) {
+	return p.getAllArchitectEmergencyGroupAttr(ctx, p)
+}
+
+func (p *architectEmergencyGroupProxy) getAllArchitectEmergencyGroup(ctx context.Context, emergencyGroupId string) (*platformclientv2.Emergencygroup, *platformclientv2.APIResponse, error) {
+	return p.getArchitectEmergencyGroupAttr(ctx, p, emergencyGroupId)
+}
+
+func (p *architectEmergencyGroupProxy) updateArchitectEmergencyGroup(ctx context.Context, emergencyGroupId string, emergencyGroup platformclientv2.Emergencygroup) (*platformclientv2.Emergencygroup, *platformclientv2.APIResponse, error) {
+	return p.updateArchitectEmergencyGroupAttr(ctx, p, emergencyGroupId, emergencyGroup)
+}
+
+func (p *architectEmergencyGroupProxy) getArchitectEmergencyGroupIdByName(ctx context.Context, name string) (emergencyGroup *platformclientv2.Emergencygrouplisting, apiResponse *platformclientv2.APIResponse, err error) {
+	return p.getArchitectEmergencyGroupIdByNameAttr(ctx, p, name)
+}
+
+func (p *architectEmergencyGroupProxy) deleteArchitectEmergencyGroup(ctx context.Context, emergencyGroupId string) (*platformclientv2.APIResponse, error) {
+	return p.deleteArchitectEmergencyGroupAttr(ctx, p, emergencyGroupId)
+}
+
+func (p *architectEmergencyGroupProxy) createArchitectEmergencyGroup(ctx context.Context, emergencyGroup platformclientv2.Emergencygroup) (*platformclientv2.Emergencygroup, *platformclientv2.APIResponse, error) {
+	return p.createArchitectEmergencyGroupAttr(ctx, p, emergencyGroup)
+}
+
+func getAllArchitectEmergencyGroupFn(ctx context.Context, p *architectEmergencyGroupProxy) (*[]platformclientv2.Emergencygroup, *platformclientv2.APIResponse, error) {
+	var totalRecords []platformclientv2.Emergencygroup
+
+	const pageSize = 100
+
+	emergencyGroupConfigs, resp, getErr := p.architectApi.GetArchitectEmergencygroups(1, pageSize, "", "", "")
+
+	if emergencyGroupConfigs.Entities == nil || len(*emergencyGroupConfigs.Entities) == 0 {
+		return &totalRecords, nil, nil
+	}
+
+	for _, emergencyGroup := range *emergencyGroupConfigs.Entities {
+		totalRecords = append(totalRecords, emergencyGroup)
+	}
+
+	for pageNum := 2; pageNum <= *emergencyGroupConfigs.PageCount; pageNum++ {
+		if getErr != nil {
+			return nil, resp, fmt.Errorf("Failed to get page of emergency group configs: %v", getErr)
+		}
+
+		if emergencyGroupConfigs.Entities == nil || len(*emergencyGroupConfigs.Entities) == 0 {
+			break
+		}
+
```

Comment:
Looks like we're missing some logic inside this for loop. There's no actual API call, and we're not appending the list of total records